port LoRDEC 0.9 to gatb-core >= 1.3.0 by replacing Graph::Vector to GraphVector
author: Kenneth Hoste (HPC-UGent)
diff -ru lordec-src_0.9.orig/lordec-correct.h lordec-src_0.9/lordec-correct.h
--- lordec-src_0.9.orig/lordec-correct.h	2018-06-04 15:44:49.000000000 +0200
+++ lordec-src_0.9/lordec-correct.h	2023-03-31 22:17:39.105922332 +0200
@@ -160,11 +160,11 @@
   // store the neighbors of begin node in the stack
 
 #if defined(GATB_V110)
-  Graph::Vector<Edge> neighbors = graph.successors<Edge>(begin);
+  GraphVector<Edge> neighbors = graph.successors<Edge>(begin);
 #elif defined(GATB_V130) || defined(GATB_V140) || defined(GATB_V141)
   GraphVector<Edge> neighbors = graph.successorsEdge(begin);
 #else
-  Graph::Vector<Edge> neighbors = graph.successorsEdge(begin);
+  GraphVector<Edge> neighbors = graph.successorsEdge(begin);
 #endif
 
   //  if (neighbors.size() == 1)
@@ -221,11 +221,11 @@
         }
       }
 #if defined(GATB_V110)
-      Graph::Vector<Edge> neighbors = graph.successors<Edge>(cnode);
+      GraphVector<Edge> neighbors = graph.successors<Edge>(cnode);
 #elif defined(GATB_V130) || defined(GATB_V140) || defined(GATB_V141)
       GraphVector<Edge> neighbors = graph.successorsEdge(cnode);
 #else
-      Graph::Vector<Edge> neighbors = graph.successorsEdge(cnode);
+      GraphVector<Edge> neighbors = graph.successorsEdge(cnode);
 #endif
 
       //      if (neighbors.size() == 1)
@@ -328,11 +328,11 @@
 
   // store the neighbors of begin node in the stack
 #if defined(GATB_V110)
-  Graph::Vector<Edge> neighbors = graph.successors<Edge>(begin);
+  GraphVector<Edge> neighbors = graph.successors<Edge>(begin);
 #elif defined(GATB_V130) || defined(GATB_V140) || defined(GATB_V141)
   GraphVector<Edge> neighbors = graph.successorsEdge(begin);
 #else
-  Graph::Vector<Edge> neighbors = graph.successorsEdge(begin);
+  GraphVector<Edge> neighbors = graph.successorsEdge(begin);
 #endif
   for (i = neighbors.size()-1; i>= 0; i--) {
     nodes.push(new stack_element(neighbors[i].to, 1, ascii(neighbors[i].nt)));
@@ -396,11 +396,11 @@
       else {
 
 #if defined(GATB_V110)
-        Graph::Vector<Edge> neighbors = graph.successors<Edge>(cnode);
+        GraphVector<Edge> neighbors = graph.successors<Edge>(cnode);
 #elif defined(GATB_V130) || defined(GATB_V140) || defined(GATB_V141)
         GraphVector<Edge> neighbors = graph.successorsEdge(cnode);
 #else
-        Graph::Vector<Edge> neighbors = graph.successorsEdge(cnode);
+        GraphVector<Edge> neighbors = graph.successorsEdge(cnode);
 #endif
         for(i=neighbors.size()-1; i>=0; i--) {
           Edge e = neighbors[i];
@@ -450,11 +450,11 @@
 
   while (true) {
 #if defined(GATB_V110)
-    Graph::Vector<Edge> neighbors = graph.successors<Edge>(node);
+    GraphVector<Edge> neighbors = graph.successors<Edge>(node);
 #elif defined(GATB_V130) || defined(GATB_V140) || defined(GATB_V141)
     GraphVector<Edge> neighbors = graph.successorsEdge(node);
 #else
-    Graph::Vector<Edge> neighbors = graph.successorsEdge(node);
+    GraphVector<Edge> neighbors = graph.successorsEdge(node);
 #endif
     if (neighbors.size() == 0) // dead end
       return DEADEND;
